packages/app-spec-automatic-chess/src/libraries/squadChess.js

```diff
@@ -0,0 +1,112 @@
+/*
+ * SQUAD CHESS RULES ENGINE
+ * 
+ * This library takes in a game state (board position, turn number, and legal moves) and an action (a move), 
+ * then returns a new state and set of legal moves.
+ * 
+ * General architecture: Take in a board state and an action (move), 
+ * return a new state and a list of legal moves.
+ * 
+ * DATA TYPES
+ * move = { from: [0,1], to: [4,6] }
+ * state = { position: {'0,1': {pieceId, player: 0||1}... }, turn: 14 }
+ * PIECES = { pieceId: { name: 'rook', mechanics: { 'move': [moveInputs] }}} (come from 'component' definitions)
+ * MECHANICS = { 'mechanic name': function that takes move inputs, returns moves } (come from 'game' definition)
+ * 
+ */
+
+const MECHANICS = {
+  'move': (inputs, from, position, turn) => {
+    // inputs = { offset, steps } // this might eventually support moves based on arbitrary formula
+      // offset = [x, y]
+    let moves = []
+    let to = from
+    // for each step
+    for(let i = 0; i < inputs.steps; i++) {
+      to = [to[0]+inputs.offset[0], to[1]+inputs.offset[1]]
+      if (!(to in position)) { break } // if off board
+      if (position[to] !== null) { break } // if not empty
+      moves.push({
+        'from': from,
+        'to': to
+      })
+    }
+    return moves
+  },
+  'capture': (inputs, from, position, turn) => {
+    let moves = []
+    let to = from
+    for(let i = 0; i < inputs.steps; i++) {
+      to = [to[0]+inputs.offset[0], to[1]+inputs.offset[1]]
+      if (!(to in position)) { break } // if off board
+      if (position[to] === null) { continue } // if empty
+      if (position[from].player === position[to].player) { break } // if same color piece
+      moves.push({
+        'from': from,
+        'to': to
+      })
+      break
+    }
+    return moves
+  }
+}
+
+let PIECES = {}
+
+const registerPieces = (pieces) => {
+  PIECES = pieces
+}
+
+const makePosition = (position, move) => {
```

Comment:
This function returns the next position after a move has been made on a given position.

I think the name `makePosition` suggests something broader than that. I'd name this something like `nextPosition` or `updatePosition`.

The term "make" suggests to me that a position is being made where there was none before.